package/run.ps1

```diff
@@ -127,10 +128,46 @@ if ($env:INSTALL_RKE2_SKIP_START -and ($env:INSTALL_RKE2_SKIP_START -eq $true))
 
 if ((Get-Service -Name $rke2ServiceName -ErrorAction SilentlyContinue)) {
     if ((Get-Service $rke2ServiceName).Status -eq 'Stopped') {
-        Write-LogInfo "Starting for RKE2 agent service"
-        Start-Service -Name $rke2ServiceName
-    } 
+        Write-LogInfo "Attempting to start the RKE2 agent service"
+        $successfulStart = $false
+        $maxAttempts = 5
+        $attempts = 1
+        # try to start the service multiple times,
+        # Start-Service calls are idempotent.
+        # In the event the rke2 service automatically
+        # restarts itself successfully during the sleep period,
+        # the try block will have no effect and we will declare a succesful start
+        # of the service. This retry loop ensures a plan will not fail
```

Comment:
This is a good point. While I didn't see any issues with starting the service this way during my testing, I did take some extra time to get a better understanding of how a windows service handles failure conditions to make sure there wouldn't be a problem. I've refactored the PR to rely more on the automatic restarting done by Windows, just to make sure we don't run into anything nasty. 

I've added a larger buffer period after each attempt at starting the service so that windows can try and restart the service itself if possible. I've found that this usually happens after in about one or two tries, which lasts between 30 and 60 seconds.
 
However, in certain cases where a host is under load and RKE2 fails to start, the service may never transition into a 'running' state which results in windows not attempting to automatically restart the service at all. So, I've kept the `for` loop to handle this scenario. I also found that a similar issue can occur with the `Restart-Service` command, wherein it fails to properly process the command and only stops the service.

The Microsoft documentation describes a [scenario where a service will unexpectedly restart if manually restarted and then manually stopped](https://learn.microsoft.com/en-us/windows/win32/api/winsvc/nf-winsvc-changeserviceconfig2a#remarks), but does not indicate that this will also happen for a service which is running when the 30 second restart period is reached, and I've not seen that behavior when looking through the logs of the service manager itself.